Project: Recursion C-like code warning

kahmoh opened the pull request:
<!-- Thank you for taking the time to contribute to The Odin Project. In order to get this pull request (PR) merged in a reasonable amount of time, you must complete this entire template. -->

## Because
<!-- Summarize the purpose or reasons for this PR, e.g. what problem it solves or what benefit it provides. -->
This solves the problem of learners getting caught up trying to understand C-like code in the Wikipedia article explaining merge sort.

## This PR
<!-- A bullet point list of one or more items describing the specific changes. -->
- Add warning note that warns learners to avoid trying to understand the code and focus on the idea it is trying to get across.

## Issue
<!--
If this PR closes an open issue in this repo, replace the XXXXX below with the issue number, e.g. Closes #2013.

If this PR closes an open issue in another TOP repo, replace the #XXXXX with the URL of the issue, e.g. Closes https://github.com/TheOdinProject/curriculum/issues/XXXXX

If this PR does not close, but is related to another issue or PR, you can link it as above without the 'Closes' keyword, e.g. 'Related to #2013'.

_Note:_ any pull request created for an issue that already has someone else assigned **will be closed without review**.
-->
Closes #27224

## Additional Information
<!-- Any other information about this PR, such as a link to a Discord discussion. -->


## Pull Request Requirements
<!-- Replace the whitespace between the square brackets with an 'x', e.g. [x]. After you create the PR, they will become checkboxes that you can click on. -->
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

JoshDevHub commented on javascript/computer_science/project_recursion.md:
```diff
@@ -20,7 +20,9 @@ Did you figure it out? Congratulations! But do you really understand what is tak
 It can be a bit strange to wrap your head around, but just remember you're "dividing and conquering" the problem.
-
+<div class="lesson-note lesson-note--warning" markdown="1">
+  The Wikipedia article above uses merge sort examples written in C-like code, this code will likely look quite different than what you're used to. Don't waste time trying to understand the specifics of the language, just focus on the idea it is trying to get across.
```
```suggestion
  The Wikipedia article above uses merge sort examples written in C-like code. This code will likely look quite different from what you're used to. Don't waste time trying to understand the specifics of the language, and just focus on the idea it is trying to get across.
```

JoshDevHub commented on ruby/computer_science/project_recursion.md:
```diff
@@ -18,6 +18,9 @@ Did you figure it out? Congratulations! But do you really understand what is tak
 
 It can be a bit strange to wrap your head around, but just remember you're "dividing and conquering" the problem.
+<div class="lesson-note lesson-note--warning" markdown="1">
+  The Wikipedia article above uses merge sort examples written in C-like code, this code will likely look quite different than what you're used to. Don't waste time trying to understand the specifics of the language, just focus on the idea it is trying to get across.
```
```suggestion
  The Wikipedia article above uses merge sort examples written in C-like code. This code will likely look quite different from what you're used to. Don't waste time trying to understand the specifics of the language, and just focus on the idea it is trying to get across.
```

JoshDevHub reviewed (changes requested):
Looks good. Thanks for doing this. And awesome that you noticed this would have to be fixed in both the JavaScript and Ruby paths.

Gonna recommend a couple small fixes with the punctuation and wording before we proceed with merging.

Also, this lesson has a few errors popping up with Markdownlint. You can see them if you click the "Files changed" tab and look through the right column of the diff. None of these were introduced by you, and they were in these files before you did anything. Would you be up for going ahead and fixing these? If not, that's okay. I can just go ahead and merge after you include my suggestions below. Just let me know.

kahmoh commented:
@JoshDevHub Just committed the wording and punctuation suggestions, gonna take a look now at some of those linting errors and see if I can figure out what's going on.

kahmoh commented:
Hi @JoshDevHub, I've just made a small few mistakes in my previous two commits, I'm going to update them and fix them, but if you would prefer that I close this PR and start a new one please let me know. Apologies for any troubles.

kahmoh commented:
I've just fixed the linting issues, and fixed the issues I made with those two bad commits. I've noticed that both affected files have a required heading structure error. However, I also noticed this was ignored in other commits to the curriculum and since I don't have access to the rules that the the linter is checking against I was unable to fix these. If there is any other issues you see here, please let me know and I'll get them sorted.
